File: leetcode/167_两数之和 II - 输入有序数组/solution.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */

function twoSum(nums, target) {
  let low = 0
  let high = nums.length - 1
  while (low < high) {
    if (nums[low] + nums[high] < target) {
      low++
    } else if (nums[low] + nums[high] > target) {
      high--
    } else {
      return [low + 1, high + 1]
    }
  }
}

console.log(twoSum([2,7,11,15], 9))

File: leetcode/167_两数之和 II - 输入有序数组/solution.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace  std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i = 0;
        int j = nums.size() - 1;
        while (i < j) {
            int sum = nums[i] + nums[j];
            if (sum < target) {
                i++;
            } else if (sum > target) {
                j--;
            } else {
                return vector<int>{i + 1, j + 1};
            }
        }
        return vector<int>{-1, -1};
    }
};

int main() {
    Solution solution;
    int a[4] = {2, 7, 11, 15};
    int target = 9;
    vector<int> nums(a, a + 4);
    vector<int> result = solution.twoSum(nums, target);

    for (int i = 0; i < result.size(); i++) {
        cout << result[i] << endl;
    }
    return 0;
}
